tests/ui/check-cfg/unknown-values.with_values.stderr

```diff
@@ -0,0 +1,7 @@
+warning: unexpected `with_values` as condition name
```

Comment:
> Aren't warnings about revision cfgs off-topic for this test?

Indeed they are. Added them manually so that no diagnostic is ever emitted.

> Maybe the revision cfgs should be added to known cfgs (possibly automatically by compiletest).

I've taught about this as well, and it's a good idea except for those tests, because having those diagnostics pop up here is very important, because with those tests we want to make sure we're properly testing what we think we're testing, and hiding things will imo only make it more likely to introduce bugs.

Reply:
Could you mark all revisition cfgs as expected?
They can be found by "was set with `--cfg` but isn't in the `--check-cfg` expected names".

It's hard to review tests due to so much noise.
The file's revision is already clearly visible from file names, so additional warnings in stderr files are not needed.

Reply:
Sure. I did it for all the tests were it wouldn't have changed the behavior; i.e. except for `exhaustive-names-values.rs`, `exhaustive-names.rs` and `exhaustive-values.rs` where expecting the revisions would be significant. All the other instances are "real" warnings.

@rustbot ready